prov/gni/src/gnix_nic.c

```diff
@@ -296,24 +504,36 @@ static int __nic_get_completed_txd(struct gnix_nic *nic,
 static int __nic_tx_progress(struct gnix_nic *nic)
```

Comment:
It might have been cleaner and faster tochange __nic_tx_progress take a CQ as a parameter.  With the reviewed design, you will continue to call __nic_get_completed_txd() on both CQs while we consistently only find TXDs on one of them.